src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNamesTests.Fixer.cs

```diff
@@ -355,6 +355,74 @@ End Sub
                 End Class");
         }
 
+        [Fact]
+        public async Task CA1071_ClassSinglePropertyPrivate_CSharp()
+        {
+            await VerifyCSharpCodeFixAsync(@"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+                    private int FirstProp { get; }
+
+                    public object SecondProp { get; }
+
+                    [JsonConstructor]
+                    public C1(int [|firstProp|], object secondProp)
+                    {
+                        this.FirstProp = firstProp;
+                        this.SecondProp = secondProp;
+                    }
+                }",
+                @"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+    public int FirstProp { get; }
+
+                    public object SecondProp { get; }
+
+                    [JsonConstructor]
+                    public C1(int firstProp, object secondProp)
+                    {
+                        this.FirstProp = firstProp;
+                        this.SecondProp = secondProp;
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task CA1071_ClassSinglePropertyPrivate_Basic()
+        {
+            await VerifyBasicCodeFixAsync(@"
+                Imports System.Text.Json.Serialization
+
+                Public Class C1
+                    Private Property FirstProp() As Integer
+                    Property SecondProp() as Object
+
+                    <JsonConstructor>
+                    Public Sub New([|firstProp|] as Integer, secondProp as Object)
+                        Me.FirstProp = firstProp
+                        Me.SecondProp = secondProp
+                    End Sub
+                End Class",
+                @"
+                Imports System.Text.Json.Serialization
+
+                Public Class C1
+    Public Property FirstProp() As Integer
```

Comment:
Should we respect "omit **Public**" (default) modifier when specified in a code style for VB? Will it work without additional modifications?

Reply:
I don't think so, that sounds like a different analyzer requirement

Reply:
@buyaa-n Thanks. Do you mean we should**n't** check **private** fields in this analyzer or just ignore the "omit **Public"** modifier?

Reply:
@psxvoid sorry, apparently I misunderstood your question previously. 

I don't think that will be respected without additional modification, but I think it is not important, user could just remove the modifier manually or run the styling fixer, so it is not required to respect that, you can ignore styling.